ConsoleApp/ConsoleApp/Program.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace ConsoleApp
+{
+
+ 
+
+
+    class Program
+    {
+
+        static void todayHoliday(IEnumerable<Holiday> holidays)
+        {
+            var sb = new StringBuilder();
+
+
+            DateTime currentDate = (DateTime.Now).Date;
+
+            var resultNext = holidays
+            .Where(x =>DateTime.Compare(currentDate, Convert.ToDateTime(x.date)) < 0)
+                .Select(x=>new
+                { 
+                  x.date,
+                  x.nameEn
+                })
+                .OrderBy(x => x.date) 
+                .First();
+
+            var resultPrevious = holidays
+            .Where(x => DateTime.Compare(currentDate, Convert.ToDateTime(x.date)) > 0)
+                .Select(x => new
+                {
+                    x.date,
+                    x.nameEn
+                })
+                .OrderByDescending(x => x.date)
+                .First();
+
+            var resultCurrent = holidays
+            .Where(x => DateTime.Compare(currentDate, Convert.ToDateTime(x.date)) == 0)
```

Comment:
instead of Where, Select and FirstOrDefault you can pass filtration predicate in FirstOrDefault